analyses/verbose/sarif_helper.go

```diff
@@ -0,0 +1,153 @@
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type RegexDependencyLocator struct {
+	FileContent         string
+	Ecosystem           string
+	EndIndices          []int
+	DependencyNodeIndex []int
+}
+
+func ProcessSarif(analysedResult *driver.GetResponseType, manifestFilePath string) (bool, error) {
+	var hasVuln bool
+	report, err := sarif.New(sarif.Version210)
+
+	if err != nil {
+		log.Fatal().Msg("Error forming SARIF respose")
+		return false, errors.New("unable to create SARIF file")
+	}
+
+	run := report.AddRun("CRDA", "https://github.com/fabric8-analytics")
+
+	regexDependencyLocator := RegexDependencyLocator{}
+	if len(analysedResult.AnalysedDeps) == 0 {
+		log.Fatal().Msg("Dependencies have not been analysed")
+		return false, errors.New("dependencies have not been analysed")
+	}
+	err = regexDependencyLocator.SetUp(manifestFilePath, analysedResult.AnalysedDeps[0].Ecosystem)
+
+	if err != nil {
+		return false, errors.New("unable to setup dependency locator")
+	}
+
+	manifestParts := strings.Split(manifestFilePath, string(os.PathSeparator))
+	manifest := manifestParts[len(manifestParts) - 1]
+	for _, dep := range analysedResult.AnalysedDeps {
+		line, column := regexDependencyLocator.LocateDependency(dep.Name)
+		for _, publicVuln := range dep.PublicVulnerabilities {
+			addVulnToReport(run, publicVuln, manifest, line, column)
+			hasVuln = true
+		}
+		for _, privateVuln := range dep.PrivateVulnerabilities {
+			addVulnToReport(run, privateVuln, manifest, line, column)
+			hasVuln = true
+		}
+	}
+
+	report.Write(os.Stdout)
+	return hasVuln, nil
+}
+
+func addVulnToReport(run *models.Run, vuln driver.VulnerabilitiesType, manifestFilePath string, line int, column int) {
+	rule := run.AddRule(vuln.ID).
+		WithHelpURI(vuln.URL).WithDescription(vuln.Title)
+
+	run.AddResult(rule.ID).
+		WithMessage(vuln.Title).
+		WithLevel(vuln.Severity).
+		WithLocationDetails(manifestFilePath, line, column)
+}
+
+
+func (r *RegexDependencyLocator) SetUp(manifestFilePath string, ecosystem string) error{
+	content, err := ioutil.ReadFile(manifestFilePath)
+	if err != nil {
+		log.Fatal().Msg("Unable to load manifest File " + manifestFilePath)
+		return fmt.Errorf("unable to load manifest file %s" ,manifestFilePath)
+	}
+
+	r.FileContent = string(content)
+	r.Ecosystem = ecosystem
+	newLineRegex, _ := regexp.Compile("\n")
+
+	lines := newLineRegex.FindAllStringIndex(r.FileContent, -1)
+	r.EndIndices = make([]int, len(lines))
+	// Finding the end index for each end of line
+	for i, line := range lines {
+		r.EndIndices[i] = line[1]
+	}
+
+	dependenciesRegex, _ := regexp.Compile(getDependencyNodeRegex(r.Ecosystem))
+	// Find the index for the start of the dependency node ( <dependencies> in case of pom.xml,
+	// dependencies: in case of package.json)
+	r.DependencyNodeIndex = dependenciesRegex.FindStringIndex(r.FileContent)
+
+	return nil
+
+}
+
+func (r *RegexDependencyLocator) LocateDependency(dependency string) (int, int){
```

Comment:
i feel `LocateDependency` is not used imported outside of this package, No point in exporting it.

It should be `locateDependency`